google/gax/auth.py

```diff
@@ -31,19 +31,49 @@
 
 from __future__ import absolute_import
 
+import grpc
```

Comment:
So this is here because eventually I'll defer most of this to `google-auth` - (see [this PR](https://github.com/GoogleCloudPlatform/google-auth-library-python/pull/67/files#diff-ce8fd607d554afd5f68f90b463e6b6eaR64) - you'll notice a similar `secure_authorized_channel` function).

I can move this into grpc, but at that point I might as well just delete the `auth` file altogether - it will provide no discernible value. Which would you like?